src/core/middleware/trailingSlash.js

```diff
@@ -10,9 +10,18 @@ export function trailingSlashesMiddleware(
   next,
   { _config = config } = {},
 ) {
+  let redirect = false;
   const UrlParts = req.originalUrl.split('?');
   const UrlSlashSeparated = UrlParts[0].replace(/^\//, '').split('/');
 
+  // Trim all params on the URL.
+  UrlSlashSeparated.forEach((param, index) => {
```

Comment:
I wonder if this should also have a condition to check it's needed prior to iterating over the path pieces? Whilst that might not make much real-world difference, this is going to be run on every incoming request.

Reply:
You mean like checking to see if any spaces exist in the URL? This also makes me think we can run `decodeURI` on the whole string before breaking it into parts, which would mean fewer calls to decode.